lib/charms/mongodb/v1/shards_interface.py

```diff
@@ -541,28 +560,102 @@ def _handle_changed_secrets(self, event) -> None:
             event.defer()
             return
 
+        # FUTURE PR: if config-server does not have TLS enabled log a useful message and go into
+        # blocked in relation_changed and other status checks
+
+    def get_membership_auth_modes(self, event: RelationChangedEvent) -> Tuple[bool, bool]:
+        """Returns the available authentication membership forms."""
+        key_file_contents = self.database_requires.fetch_relation_field(
+            event.relation.id, KEYFILE_KEY
+        )
+        tls_ca = self.database_requires.fetch_relation_field(event.relation.id, INT_TLS_CA_KEY)
+        return (key_file_contents is not None, tls_ca is not None)
+
+    def update_member_auth(
+        self, event: RelationChangedEvent, membership_auth: Tuple[bool, bool]
+    ) -> None:
+        """Updates the shard to have the same membership auth as the config-server."""
+        cluster_auth_keyfile, cluster_auth_tls = membership_auth
+        tls_integrated = self.charm.model.get_relation(Config.TLS.TLS_PEER_RELATION)
+
+        # Edge case: shard has TLS enabled before having connected to the config-server. For TLS in
+        # sharded MongoDB clusters it is necessary that the subject and organisation name are the
+        # same in their CSRs. Re-requesting a cert after integrated with the config-server
+        # regenerates the cert with the appropriate configurations needed for sharding.
+        if cluster_auth_tls and tls_integrated and not self._should_request_new_certs():
```

Comment:
maybe we should `request_certificate()` if `_should_request_new_certs==True`
```suggestion
        if cluster_auth_tls and tls_integrated and self._should_request_new_certs():
```